<a href="https://colab.research.google.com/github/purnawanpp/plants/blob/main/plants.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Protein-Ligand ANTSystem (PLANTS) simulation by Purnawan Pontana Putra**

**Required software installation**

In [ ]:
!wget https://github.com/purnawanpp/plants/raw/main/plants
!wget https://github.com/purnawanpp/plants/raw/main/spores
!apt install openbabel
!chmod +x /content/plants
!chmod +x /content/spores
!wget https://github.com/purnawanpp/plants/raw/main/dock_gcolab.txt

--2023-03-08 10:52:44--  https://github.com/purnawanpp/plants/raw/main/plants
Resolving github.com (github.com)... 20.27.177.113
Connecting to github.com (github.com)|20.27.177.113|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/purnawanpp/plants/main/plants [following]
--2023-03-08 10:52:45--  https://raw.githubusercontent.com/purnawanpp/plants/main/plants
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 9536847 (9.1M) [application/octet-stream]
Saving to: ‘plants.1’

plants.1            100%[===================>]   9.09M  --.-KB/s    in 0.09s   

2023-03-08 10:52:45 (96.4 MB/s) - ‘plants.1’ saved [9536847/9536847]

--2023-03-08 10:52:45--  https://github.com/purnawanpp/plants/raw/main/spores
R

**Download Protein PDB ID 3HTB**

In [ ]:
!wget https://files.rcsb.org/download/3HTB.pdb

--2023-03-08 10:52:53--  https://files.rcsb.org/download/3HTB.pdb
Resolving files.rcsb.org (files.rcsb.org)... 128.6.158.70
Connecting to files.rcsb.org (files.rcsb.org)|128.6.158.70|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/octet-stream]
Saving to: ‘3HTB.pdb’

3HTB.pdb                [ <=>                ] 161.13K  --.-KB/s    in 0.06s   

2023-03-08 10:52:53 (2.61 MB/s) - ‘3HTB.pdb’ saved [164997]



**Separete Protein and ligand**

In [ ]:
!./spores --mode splitpdb 3HTB.pdb


				 SPORES        

	      Structure PrOtonation and REcognition System

			      version 1.3


author: T. ten Brink

scientific contributors: P. Duchstein, J. Gimmler, O. Korb, M. Zerara,
			  T.E. Exner

contact: tim.tenbrink@uni-konstanz.de


molecule number: 1
64 alternative atomic positions have been removed !
splitting 3HTB
225
protein: 0
ligs: 4
water: 660
metal: 0


**Add Hydrogen, charge (gastaiger) dan protonasi pada pH 7.4**

Information:

⚾!obel: command to run the Obabel program on Google Colab Notebook.

⚾-imol2: option to specify the input file format, i.e. mol2.

⚾/content/protein.mol2: path of the input file to be converted and manipulated.

⚾-omol2: option to specify the output file format, i.e. mol2.

⚾-h: option to add hydrogen to the molecule.

⚾-p 7.4: option to add charge to a molecule with a pH of 7.4.

⚾-b: option to add hydrogen bond information.

⚾--ff "gasteiger": option to specify the calculation force, in this case using the Gasteiger method.

⚾-O rec.mol2: option to specify the name and location of the output file, i.e. rec.mol2




In [ ]:
!obabel -imol2 protein.mol2 -omol2 -h -b --ff "gasteiger" -O rec.mol2
!obabel -imol2 ligand_JZ4167_0.mol2 -omol2 -h -b --ff "gasteiger" -O lig.mol2

*** Open Babel Warning  in ReadMolecule
  Failed to kekulize aromatic bonds in MOL2 file (title is 3HTB_protein)

1 molecule converted
1 molecule converted


**Create and define binding_site or Gridbox**

In [ ]:
!./plants --mode bind lig.mol2 rec.mol2



                  PLANTS
        Protein-Ligand ANT System
               version 1.2


author: Oliver Korb

scientific contributors: T.E. Exner, T. Stuetzle

contact: Oliver.Korb@uni-konstanz.de


run PLANTS: PLANTS --mode screen yourconfigfile



Calculated bindingsite definition (from 1 ligands): 

bindingsite_center 22.7142 -25.2627 -3.283
bindingsite_radius 3.75624



Move data:

* bindingsite_center 22.7142 -25.2627 -3.283
* bindingsite_radius 3.75624

into the file **dock_gcolab.txt**

**Molecular Docking Simulation**

In [ ]:
!./plants --mode screen /content/dock_gcolab.txt



                  PLANTS
        Protein-Ligand ANT System
               version 1.2


author: Oliver Korb

scientific contributors: T.E. Exner, T. Stuetzle

contact: Oliver.Korb@uni-konstanz.de


run PLANTS: PLANTS --mode screen yourconfigfile

PLANTS info: CHO hydrogen found 513 HD2 HIS3131
PLANTS info: CHO hydrogen found 514 HE1 HIS3131
virtual screening progress: 1 of 1
current ligand: /content/lig.mol2 (entry 1)
LIGAND DOFs: 9
PROTEIN DOFs: 0
Simplex dimension: 9
Simplex dimension: 9
iterations : 187
starting optimization ...
problem dimension: 9
ATOMS / s: 8.86805e+06
EVAL / s: 886805
optimization finished after 0.82s
best score: -67.44


total virtual screening time: 0.98s
Ligands skipped due to errors: 0


**The ΔG scores or docking results**

After the process is complete, don't forget to click the Google Colab refresh box/file icon. Docking simulation results can be seen in the *results* folder with the file name **bestranking.csv**, see energy TOTAL_SCORE is a delta G value

In [ ]:
import pandas as pd
df = pd.read_csv('/content/results/bestranking.csv')
df.head()

LIGAND_ENTRY  TOTAL_SCORE  SCORE_RB_PEN  \
0  3HTB_ligand_2_entry_00001_conf_01     -67.4383      -61.4383   

   SCORE_NORM_HEVATOMS  SCORE_NORM_CRT_HEVATOMS  SCORE_NORM_WEIGHT  \
0             -6.74383                 -31.3021          -0.495193   

   SCORE_NORM_CRT_WEIGHT  SCORE_RB_PEN_NORM_CRT_HEVATOMS  SCORE_NORM_CONTACT  \
0               -13.1078                        -28.5172            -6.74383   

     EVAL      TIME  
0  729961  0.823136

Explanation:

* LIGAND_ENTRY: Name or identification code of the ligand that was docked.
* TOTAL_SCORE: Overall score generated from docking the ligand to the target protein.
* SCORE_RB_PEN: Score for rotamer penalty.
* SCORE_NORM_HEVATOMS: Score for heavy heteroatoms (in this case, normalized value).
* SCORE_NORM_CRT_HEVATOMS: Score for normalized heavy heteroatoms with correction.
* SCORE_NORM_WEIGHT: Score normalized by the weight of the ligand molecule.
* SCORE_NORM_CRT_WEIGHT: Score normalized by correction and weight of the ligand molecule.
* SCORE_RB_PEN_NORM_CRT_HEVATOMS: Score for normalized rotamer and heavy heteroatoms with correction.
* SCORE_NORM_CONTACT: Score for normalized contact.
* EVAL: Evaluation value that measures the quality of docking between the ligand and target protein.
* TIME: Time required to perform the docking.

**OPTIONAL-View the file namely ranking.csv file. It is same as bestranking.csv, but for all ligand poses**

In [ ]:
from IPython.display import HTML
import pandas as pd

df = pd.read_csv('/content/results/ranking.csv')

html_table = df.to_html(index=False)
display(HTML(html_table))

**Calculation of RMSD simulation (re-docking)**

In [ ]:
!obrms -f lig.mol2 /content/results/3HTB_ligand_2_entry_00001_conf_01.mol2 | sed 's/3HTB_ligand_2:3HTB_ligand_2_entry_00001_conf_01/RE-DOCKING/g'

RMSD RE-DOCKING 0.434581
